src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorProjectEngine.cs

```diff
@@ -1,97 +1,329 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.Language.Components;
 using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Razor.PooledObjects;
 
 namespace Microsoft.AspNetCore.Razor.Language;
 
-public abstract class RazorProjectEngine
+public class RazorProjectEngine
```

Comment:
```suggestion
public sealed class RazorProjectEngine
```

Believe this type can be sealed now.

Reply:
It can't be sealed. `SourceGeneratorProjectEngine` inherits from it.